### PR TITLE
Wip #12417 selection

### PR DESCRIPTION
**Pull request by man-axelor**

### CODE REVIEW

File: axelor-base/src/main/resources/views/Partner.xml [hunk 1]
```diff
@@ -727,7 +727,9 @@
 	</action-attrs>
 
 	<action-record name="action-partner-record-set-balance" model="com.axelor.apps.base.db.Partner">
-		<field name="$balance" expr="eval: __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceDueCustAccount }"/>
+        <field name="$balance" expr="eval: __config__.app.getApp('account')?.partnerBalanceViewerConfiguration == 1 ? __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceCustAccount } : __config__.app.getApp('account')?.partnerBalanceViewerConfiguration == 2 ?
+        __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceDueCustAccount } :
+        __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceDueDebtRecoveryCustAccount }"/>
```
Comment (gdu-axelor) [inline axelor-base/src/main/resources/views/Partner.xml]: On Partner, should be done in JEL not groovy to be consistent.

File: axelor-account/src/main/resources/domains/AppAccount.xml [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 		<boolean name="printReportOnVentilation" title="Print report on invoice ventilation" default="true" />
 
+        <integer name="partnerBalanceViewerConfiguration" selection="account.partner.bal.config.select" default="1"/>
```
Comment (gdu-axelor) [inline axelor-account/src/main/resources/domains/AppAccount.xml]: name should be reviewed

File: axelor-base/src/main/resources/views/Partner.xml (hunk 1 above)
Comment (man-axelor) [reply axelor-base/src/main/resources/views/Partner.xml]: There is no sum function in JEL. Should we create a service to be called to do the sum ?

File: axelor-base/src/main/resources/views/Partner.xml [hunk 3]
```diff
@@ -726,9 +726,15 @@
...
-	<action-record name="action-partner-record-set-balance" model="com.axelor.apps.base.db.Partner">
-		<field name="$balance" expr="eval: __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceDueCustAccount }"/>
-	</action-record>
+    <action-record name="action-partner-record-set-balance"
+        model="com.axelor.apps.base.db.Partner">
+        <field name="$balance"
+            expr="#{accountingSituationList.stream().map(
+                    __config__.app.getApp('account').partnerBalConfigSelect == 1 ? (x -> x.balanceCustAccount)
+                    : __config__.app.getApp('account').partnerBalConfigSelect == 2 ? (x -> x.balanceDueCustAccount)
+                    : (x -> x.balanceDueDebtRecoveryCustAccount)
```
Comment (gdu-axelor) [inline axelor-base/src/main/resources/views/Partner.xml]: Default value should be "1" Total balance, not recovery balance.

**Pull request merged**